x-pack/plugins/uptime/public/hooks/use_monitor_with_in_flight_alert.ts

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useSelector } from 'react-redux';
+import { pendingSimpleAlertCreationSelector } from '../state/selectors';
+
+export const useMonitorWithInFlightAlert = (monitorId: string): boolean => {
+  const monitorsWithInFlightRequests = useSelector(pendingSimpleAlertCreationSelector);
+
+  return monitorsWithInFlightRequests.some((f) => f.monitorId === monitorId);
```

Comment:
I'm pretty sure it does short-circuit. According to [Mozilla docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some):

> The some() method executes the callback function once for each element present in the array until it finds the one where callback returns a truthy value (a value that becomes true when converted to a Boolean).
If such an element is found, some() immediately returns true. Otherwise, some() returns false.